server/games/game.py

```diff
@@ -533,7 +543,11 @@ def clear_slot(self, slot_index):
         if self.gameOptions['Victory'] != Victory.DEMORALIZATION and self.game_mode != 'coop':
             await self.mark_invalid(ValidityState.WRONG_VICTORY_CONDITION)
 
-        if self.is_ffa:
+        if self.has_ai:
+            await self.mark_invalid(ValidityState.HAS_AI)
+        elif self.is_multi_team:
+            await self.mark_invalid(ValidityState.MULTI_TEAM)
+        elif self.is_ffa:
             await self.mark_invalid(ValidityState.FFA_NOT_RANKED)
         elif not self.is_even:
             await self.mark_invalid(ValidityState.UNEVEN_TEAMS_NOT_RANKED)
```

Comment:
calling self.mark_invalid looks a bit duplicated to me/verbose, maybe consider (I'm not quite sure though) replacing that.

Is there some way to create a mapping between the variables and Validity states and then just get the validity state into a local variable that you then:

```
if(invalidState != unkownNullThingy) {
   await self.mark_inv(invalidState)
}
```